src/components/ActionButton/DisableableToggleActionButton.ts

```diff
@@ -0,0 +1,120 @@
+    static ID = 'DisableableToggleActionButton';
+    static ACTIVATED_CLASS_NAME = 'coveo-toggleactionbutton-activated';
+
+    private innerStatefulActionButton: StatefulActionButton;
+    private activatedState: ActivatedState;
+    private deactivatedState: UnactivatedState;
+    private disabledState: DisabledState;
+
+    static options: IDisableableToggleActionButtonOptions = {
+        ...ToggleActionButton.options,
+        disabledTooltip: ComponentOptions.buildStringOption(),
+        disabledIcon: ComponentOptions.buildStringOption(),
+    };
+
+    constructor(public element: HTMLElement, public options: IDisableableToggleActionButtonOptions, public bindings?: IResultsComponentBindings) {
+        super(element, DisableableToggleActionButton.ID, bindings);
+        this.options = ComponentOptions.initComponentOptions(element, DisableableToggleActionButton, options);
+
+        this.createInnerButton(bindings);
+    }
+
+    /**
+     * Indicates whether the toggle button is in the activated state.
+     */
+    public isActivated(): boolean {
+        return this.innerStatefulActionButton.getCurrentState() === this.activatedState;
+    }
+
+    /**
+     * Indicates whether the disableable toggle button is in the disable state.
+     */
+    public isDisabled(): boolean {
+        return this.innerStatefulActionButton.getCurrentState() === this.disabledState;
+    }
+
+    /**
+     * Sets the toggle button to the specified state.
+     * @param activated Whether the button is activated.
+     */
+    public setActivated(activated: boolean): void {
+        if (this.isDisabled() && !activated) {
+            this.innerStatefulActionButton.switchTo(this.deactivatedState);
+        }
+        if (!this.isDisabled() && activated !== this.isActivated()) {
+            this.innerStatefulActionButton.switchTo(activated ? this.activatedState : this.deactivatedState);
+        }
+    }
+
+    public setEnabled(enabled: boolean): void {
+        if (enabled) {
+            this.enable();
+        } else {
+            this.disable();
+        }
+    }
+
+    public disable(): void {
+        if (this.isDisabled()) {
+            return;
+        }
+        if (this.isActivated()) {
+            this.innerStatefulActionButton.switchTo(this.deactivatedState);
+        }
+        this.innerStatefulActionButton.switchTo(this.disabledState);
+    }
+
+    public enable(): void {
+        if (this.isDisabled()) {
+            this.innerStatefulActionButton.switchTo(this.deactivatedState);
+        }
+    }
+
+    public onClick(): void {
+        if (this.isDisabled()) {
+            return;
+        }
+        this.setActivated(!this.isActivated());
+
+        if (this.options.click) {
+            this.options.click();
+        }
+    }
+
+    private createInnerButton(bindings?: IResultsComponentBindings): void {
+        this.deactivatedState = new UnactivatedState(this);
```

Comment:
I thought about it, but when reviewing the class itself, it made less sense I think because the UnactivatedState is more like a 'neutral' state (neutral in the sense shift box)
I thought this reflects better the fact that there are no entry or exit functions for this state.

But both seems fine for me tho.